server/task/task-memory/src/test/java/org/apache/james/task/SerialTaskManagerWorkerTest.java

```diff
@@ -225,6 +225,10 @@ void theWorkerShouldCancelAnInProgressTask() throws InterruptedException {
 
         resultMono.block(Duration.ofSeconds(10));
 
+        // Due to the use of signals, cancellation cannot be instantaneous
+        // Let a grace period for the cancellation to complete to increase test stability
+        Thread.sleep(50);
```

Comment:
Wouldn't an assert until be better here?

Reply:
Not really, because the method can (apparently) be called more than once, but not systematically: we don't know what to await for.

Reply:
i am not convinced by this fix as we block on the result Mono the line before and in case of cancellation.
the result mono is this : 
```
 Mono.from(listener.cancelled(taskWithId.getId(), taskWithId.getTask().details()))
                .then(Mono.empty());
```
so i see not point in adding a wait since after the block the listener.cancelled would have been called.
Furthermore when launching this test until failure with intellij it doesn't fail on my local machine even after 3000+ runs.

Maybe the instabiblity is coming from another test ?
How did you reproduced the issue ?